File: 357. Count Numbers with Unique Digits.py
class Solution(object):
    def countNumbersWithUniqueDigits(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 0:
            return 1
        if n == 1:
            return 10
        res = 10
        for i in xrange(1,n):
            temp = 9
            for j in xrange(9, 9 - i, -1):
                temp *= j
            res += temp
        return res

File: 357. Count Numbers with Unique Digits.cpp
class Solution {
public:
    int countNumbersWithUniqueDigits(int n) {
        if (n == 0) return 1;
        if (n == 1) return 10;
        int res = 10;
        for (int i = 1; i < n; i++)
        {
            int temp = 9;
            for (int j = 9; j > 9-i; j--)
                temp *= j;
            res += temp;
        }
        return res;
    }
};
